### PR TITLE
Variables related fixes

### PR DESCRIPTION
**Pull request by thuck**
This includes the check of the $ for the variable assigned.
So set something /home generates an error, since the correct syntax must be set $something /home.
This also should fix partially the issue #681 since now it handles the undefined behavior of p.we_wordv.

### CODE REVIEW

File: sway/commands.c [hunk 1]
```diff
@@ -2152,6 +2152,10 @@ static struct cmd_results *cmd_set(int argc, char **argv) {
 		return error;
 	}
 
+    if (argv[0][0] != '$') {
+        return cmd_results_new(CMD_FAILURE, "set", "Malformed variable assignment, name has to start with $");
+    }
```
Comment (SirCmpwn) [inline sway/commands.c]: Indentation issues here.

**Comment by SirCmpwn** [comment]
I would rather add the $ for them than raise an error.

**Comment by thuck** [comment]
From what I understood from the i3 code this is the default behavior, it only raises an error.
But I can change this if you think it would be more aligned with sway.

**Comment by SirCmpwn** [comment]
I think it would be better, yes. Log a warning though.

**Comment by thuck** [comment]
@SirCmpwn is it ok now? Or do you need any modification to merge this?

**Comment by SirCmpwn** [comment]
Thanks!

**Pull request merged**